crates/tiny/src/conn.rs

```diff
@@ -544,6 +544,37 @@ fn handle_irc_msg(ui: &UI, client: &dyn Client, msg: wire::Msg) {
                     &format!("{} is away: {}", nick, msg),
                     &MsgTarget::User { serv, nick },
                 );
+            }
+            // ERR_BADCHANNAME
+            else if n == 479 && n_params > 2 {
+                let chan = &params[1];
+                let msg = &params[2];
+                ui.add_client_err_msg(
+                    &format!("Unable to join {}: {}", chan, msg),
+                    &MsgTarget::Chan {
+                        chan: ChanNameRef::new(chan),
+                        serv,
+                    },
+                );
+            }
+            // ERR_CHANNELISFULL    471
+            // ERR_INVITEONLYCHAN   473
+            // ERR_BANNEDFROMCHAN   474
+            // ERR_BADCHANNELKEY    475
+            // ERR_NEEDREGGEDNICK   477
+            // ERR_THROTTLE         480
+            else if (n == 471 || n == 473 || n == 474 || n == 475 || n == 477 || n == 480)
+                && n_params == 3
+            {
+                let chan = &params[1];
+                let msg = &params[2];
+                ui.add_client_err_msg(
+                    msg,
+                    &MsgTarget::Chan {
+                        chan: ChanNameRef::new(chan),
+                        serv,
+                    },
+                );
```

Comment:
Revert unrelated changes please. Didn't you already revert some of this file? Why keep this?

Happy to review these changes in their own PR.

Reply:
These are actually needed for this PR.  Since it is necessary to create the tab from`/join`(in order to set the tab configs) BEFORE the actual JOIN message is sent over the wire, we have to support these for when you can't join the channel. It also adds a nicer UX, in my opinion, when you can't join a channel that you added to your config file. It will spawn the tab but tell you that the reason why you couldn't join.

I'm a little lost in this whole feature's design/implementation now, since I could probably get around this issue by storing the tab config on the tui...can be like this:

Right now it's like:
1. `/join #blah -ignore`
2. spawns #blah tab and sets ignored
3. JOIN sent over the wire 
4. Joined or might be an error so show it on the tab

Can be like this instead:
1. /join #blah -ignore
2. triggers `ui.set_tab_config()`...stores a TabConfig for blah channel on the TUI (probably will be storing configs anyway)
3. JOIN sent over the wire
4. #blah tab is spawned since that's what happens anyway

Reply:
OK, I see the motivation now. I still think these should be in a separate PR though. Reasons:

- These messages should be handled even without this PR. How do we show these errors currently? I hope we're not just ignoring them.
- This PR would work without handling these messages. UX would not be ideal, but I think it also wouldn't be any worse than before. Only case where without these changes this PR would make things worse is if we currently crate a tab and show these errors there, but I doubht that is the case.

Do you know what we do for these messages currently, @trevarj?

Reply:
They get logged but just ignored. I am working on cleaning this up.

Reply:
> They get logged but just ignored. I am working on cleaning this up.

OK, this is bad. Thanks for working on this.

Btw, another similar issue is #53. I don't remember how bans work on IRC but if there are ban messages we should also support them while we're at it.

Reply:
@trevarj Could you move this change to its own PR please? It's useful on its own, we should just merge it. That will also make reviewing the PR easier.